api/extension-guides/language-model.md

```diff
@@ -19,35 +19,100 @@ The process for using the Language Model API consists of the following steps:
 
 ## Links
 
 - [Chat extension sample](https://github.com/microsoft/vscode-extension-samples/tree/main/chat-sample)
 - [LanguageModels API](https://github.com/microsoft/vscode/blob/5d6671dacb9d6a582b9354ea317211a8e2b2f918/src/vscode-dts/vscode.d.ts#L19190)
-- [GitHub Copilot Trust Center](https://resources.github.com/copilot-trust-center/)
+- [@vscode/prompt-tsx npm package](https://www.npmjs.com/package/@vscode/prompt-tsx)
+
+## Build the language model prompt
+
+To interact with a language model, extensions should first craft their prompt, and then send a request to the language model. You can use prompts to provide instructions to the language model on the broad task that you're using the model for. Prompts can also define the context in which user messages are interpreted.
+
+The Language Model API supports two approaches for building a language model prompt:
+
+- Use the `LanguageModelChatMessage` class to create the prompt by providing one or more user messages as strings
+- Use [`@vscode/prompt-tsx`](https://www.npmjs.com/package/@vscode/prompt-tsx) to declare the prompt by using the TSX syntax
 
-## Prompt crafting
+You can use the `prompt-tsx` library if you want more control over how the language model prompt is composed. For example, you can assign a priority to each chat message to dynamically adapt to each language model's context window size. If a rendered prompt has more message tokens than can fit into the available context window, the prompt renderer prunes messages with the lowest priority, while preserving the message order. Learn more about [`@vscode/prompt-tsx`](https://www.npmjs.com/package/@vscode/prompt-tsx).
 
-To interact with a language model, extensions should first craft their prompt and then send a request to the language model. You can use prompts to provide instructions to the language model on the broad task that you're using the model for. Prompt can also define the context in which user messages are interpreted.
+To learn more about the concepts of prompt engineering, we suggest reading OpenAI's excellent [Prompt engineering guidelines](https://platform.openai.com/docs/guides/prompt-engineering).
 
-In the following example, the first message is used to specify the persona used by the model in its replies and what rules the model should follow.
-The second message then provides the specific request or instruction coming from the user. It determines the specific task to be accomplished.
+>**Tip:** take advantage of the rich VS Code extension API to get the most relevant context and include it in your prompt. For example, to include the contents of the active file in the editor.
+
+### Use the `LanguageModelChatMessage` class
+
+The Language Model API provides the `LanguageModelChatMessage` class to create user messages. A language model prompt consists of one or more user messages that define the context in which the user's message is interpreted.
```

Comment:
And also here, it shouldn't sound like _only_ user messages. We might want to add a sentence about how creating System messages isn't supported, but might be in the future?

Reply:
Also maybe we should add history or examples to the chat-sample so we're using Assistant messages

Reply:
Added info about which types of messages we support (user & assistant). It might be good to include chat history to the sample and also include this in the extension guide.